subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -351,27 +355,40 @@ static uint8_t ticker_by_slot_get(struct ticker_node *node, uint8_t ticker_id_he
 #endif /* CONFIG_BT_TICKER_LOW_LAT */
 
 /**
- * @brief Get next ticker with slot ticks
+ * @brief Get next ticker with slot ticks or match
  *
- * @details Gets the next ticker which has slot ticks specified and
- * return the ticker id and accumulated ticks until expiration. If no
- * ticker nodes have slot ticks, the next ticker node is returned.
- * If no head id is provided (TICKER_NULL) the first node is returned.
+ * @details Iterates ticker nodes from ticker_id_head. If no head id is provided
+ * (TICKER_NULL), iteration starts from the first node.
+ * Operation details:
+ *
+ * NORMAL MODE (!CONFIG_BT_TICKER_SLOT_AGNOSTIC)
+ * - Gets the next ticker which has slot ticks specified and return the ticker
+ *   id and accumulated ticks until expiration.
+ * - If a matching function is provided, this function is called and node iteration
+ *   continues until match function returns true.
+ *
+ * SLOT AGNOSTIC MODE (CONFIG_BT_TICKER_SLOT_AGNOSTIC)
+ * - Gets the next ticker node.
+ * - If a matching function is provided, this function is called and node iteration
+ *   continues until match function returns true.
  *
  * @param instance          Pointer to ticker instance
  * @param ticker_id_head    Pointer to id of first ticker node [in/out]
  * @param ticks_current     Pointer to current ticks count [in/out]
  * @param ticks_to_expire   Pointer to ticks to expire [in/out]
- *
+ * @param fp_match_op_func  Pointer to match function or NULL if unused
+ * @param match_op_context  Pointer to operation context passed to match
+ *                          function or NULL if unused
+ * @param lazy              Pointer to lazy variable to receive lazy_current
+ *                          of found ticker node
  * @internal
  */
 static void ticker_by_next_slot_get(struct ticker_instance *instance,
 				    uint8_t *ticker_id_head, uint32_t *ticks_current,
-#if defined(CONFIG_BT_TICKER_LAZY_GET)
-				    uint32_t *ticks_to_expire, uint16_t *lazy)
-#else /* !CONFIG_BT_TICKER_LAZY_GET */
-				    uint32_t *ticks_to_expire)
-#endif /* !CONFIG_BT_TICKER_LAZY_GET */
+				    uint32_t *ticks_to_expire,
+				    ticker_op_match_func fp_match_op_func,
+				    void *match_op_context,
+				    uint16_t *lazy)
```

Comment:
~Any reason to not conditional compile the parameters?~
Ok, I find that the constant `NULL` is passed in the calls to `ticker_by_next_slot_get` and compile should optimize out the code associated with these parameters.